cassandra/src/main/java/org/apache/calcite/adapter/cassandra/CassandraFilter.java

```diff
@@ -174,14 +180,26 @@ private String translateMatch(RexNode condition) {
       }
     }
 
-    /** Convert the value of a literal to a string.
+    /** Returns the value of the literal.
      *
      * @param literal Literal to translate
-     * @return String representation of the literal
+     * @return The value of the literal in the form of the actual type.
      */
-    private static String literalValue(RexLiteral literal) {
-      Object value = literal.getValue2();
-      return String.valueOf(value);
+    private static Object literalValue(RexLiteral literal) {
+      Comparable value = RexLiteral.value(literal);
+      switch (literal.getTypeName()) {
+      case TIMESTAMP:
+      case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+        assert value instanceof TimestampString;
+        final SimpleDateFormat dateFormatter =
+            getDateFormatter(TIMESTAMP_FORMAT_STRING2);
+        return dateFormatter.format(literal.getValue2());
+      case DATE:
+        assert value instanceof DateString;
+        return value.toString();
+      default:
+        return literal.getValue3();
+      }
```

Comment:
Why change from `getValue2()` to `getValue3()` ?

Reply:
Comparing with getValue2, getValue3 add the processing for type DECIMAL, not much difference. And IMO, getValue3 covered more situations